tracer/src/Datadog.Trace/Ci/Test.cs

```diff
@@ -38,7 +38,7 @@ internal Test(TestSuite suite, string name, DateTimeOffset? startDate, TraceId t
         var module = suite.Module;
 
         var tags = new TestSpanTags(Suite.Tags, name);
-        var tracer = Tracer.Instance;
```

Comment:
I don't know if I'm understanding this correctly I think is safe to change it to `Tracer.InternalInstance` due the fact that all CI Visibility events must be root spans (yeah, Test, Suites, Modules and Sessions are root spans) so we don't need to capture a shared context here. 

In case of the integration tests I guess the automatic instrumentation will be capturing this context?

Reply:
I think the point is, will _customers_ be invoking this code? i.e. if they reference the package, will they directly/indirectly call this? If yes, then I think we need to use `Instance`, otherwise they'll access the wrong tracer. 

The point is, we must _never_ call `Tracer.InstanceInternal` if we're invoking from a _potential_ version conflict situation

Reply:
uhmm ok, well yeah this is a public api, if customers wants to create a manual integration for a custom testing framework, etc... they would need to reference the nuget package and call these methods. But also, we dogfood this API meaning that our automatic instrumentation uses the API for the supported testing frameworks.

Reply:
@zacharycmontoya you may have to weigh in here, I still think it needs to be `Instance` but I'll defer to you on that one 😄

Reply:
Looking more closely at this, I think we can actually use the internal tracer in these callsites.

The good thing is that all of these calls are from the constructors, so as long as we don't allow customers to invoke them directly, but hide them behind factory methods, we can also proxy these across the manual boundary. For this PR, let's keep these as using `Tracer.InstanceInternal` because my current understanding is that it's safe.

I know there's a heck of a lot of proxying that I've hand waved in some of the PR comments, so my next focus is going to be showing the potential API changes which would facilitate this work, because if we don't agree on the resulting API experience, we can't implement the solution.